### PR TITLE
fix: allow patch request in the case when the session is being created

### LINKED ISSUE #5836: Can't Submit a session via cfs
**Describe the bug**

I cannot submit a session via cfs by a user who has **not** created the event
**To Reproduce**
1. Go to CFS of an event which user has not created
2. Click on the add session proposal
3. Fill the details and hit submit

**Expected behavior**

It should not be happening, the session should've been accepted

**Screenshots**

ezgif com-video-to-gif 3

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Ubuntu 16.04 LTS
 - Browser [e.g. chrome, safari] Chrome

**Additional context**

It's up for grabs but would love it if someone confirms it too

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #5836

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Allows the user to patch a session in case he is not admin upon session creation. Once the session has been created, no other non-admin user will be able to access the resource as requested.

#### Changes proposed in this pull request:

- Allow patch in case the session has no speakers

### CODE REVIEW

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Please review!

File: app/api/helpers/permission_manager.py [hunk 1]
```diff
@@ -154,9 +154,13 @@ def is_speaker_for_session(view, view_args, view_kwargs, *args, **kwargs):
         return view(*view_args, **view_kwargs)
 
     if session.speakers:
+        # If the session is being edited
         for speaker in session.speakers:
             if speaker.user_id == user.id:
                 return view(*view_args, **view_kwargs)
+    else:
+        # If the session is being created
+        return view(*view_args, **view_kwargs)
```
Comment (iamareebjamal) [inline app/api/helpers/permission_manager.py]: This will allow access for all other cases

Comment (uds5501) [reply app/api/helpers/permission_manager.py]: @iamareebjamal only in the sessions which have no speakers. but to register a session, we need to have at least one speaker. right?

Comment (iamareebjamal) [reply app/api/helpers/permission_manager.py]: Confirm if this is true

Comment (uds5501) [reply app/api/helpers/permission_manager.py]: @iamareebjamal I will try and make a session via api and get back to you

Comment (uds5501) [reply app/api/helpers/permission_manager.py]: @iamareebjamal No, my observations were incorrect. I will come up with another solution in that case.

**Comment by uds5501** [comment]
@iamareebjamal to assign a speaker to a session via cfs, a patch request is required. before that, the session is not allocated to any particular user. So, in that case, how do I enforce that this session is of the submitting user only?

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Please review

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Please review :)

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Please review :)

**Comment by uds5501** [comment]
@CosmicCoder96 Please review :)

**Pull request merged**